### PR TITLE
feat(vault): pending withdrawal ui

### PR DESCRIPTION
**Pull request by gbarkhatov**
## What this PR does

It reworks the "Pending Withdraw" card on the dashboard so a withdrawal shows its real progress as a series of clear stages — Submitted → In progress → Challenge period → Payout sent — plus a Blocked error state. Each card now has a progress bar, the withdrawal's own transaction hash, the date, and (during the challenge period) a live "time remaining" estimate and a short security note. This is a UI-only change: all the data was already available, it just wasn't being shown.

## Why

The old card was misleading. It implied a withdrawal takes "a few hours", but a withdrawal actually goes through a Bitcoin challenge period of about three days before the payout can be sent. It also led with the original peg-in transaction hash, which is not the withdrawal transaction. Team for a staged view that tells the user honestly where their withdrawal is and how long is left, and protocol confirmed the timing comes from on-chain contract parameters (so it is fetched, not hardcoded).

## What changed

- New card component `PendingWithdrawCard` that renders one withdrawal as a staged card (status badge with an info tooltip, progress bar, single TX hash, date, and the challenge-period note + countdown).
- Stage model lives in `pegoutStateMachine.ts`: the existing claimer status is mapped to the five stage labels, and a new helper computes the progress-bar fill (the challenge period fills gradually as the assert transaction gains confirmations).
- The single "TX Hash" shows the withdrawal's own transaction — the claim tx once it is broadcast, then the assert tx during and after the challenge period, and a "Pending" placeholder before anything is on-chain.
- The dashboard no longer hides a withdrawal the moment its payout is broadcast, so the "Payout sent" stage is actually visible. The card then drops off on its own once the vault leaves the redeemed set on-chain.
- Copy moved into `copy.ts` (stage labels, "Est. Remaining", the challenge-period note, "Contact Support"). External links (docs + support) are named constants.
- Removed the now-unused `PegoutTxHashRow` component and its copy (no dead code left behind).

## Approach, and why we chose it

- Build a dedicated card instead of extending the shared `VaultDetailCard`. The shared card is also used by deposits and has a fixed layout. The new design (progress bar under the header, challenge-period note, Contact Support button) does not fit it cleanly, so a separate component keeps both cards simple and avoids piling conditional props onto shared code. We still reuse the small building blocks (card shell, rows, copyable hash, progress bar, status badge), so nothing is reinvented.
- Drive the "Blocked" treatment off the real protocol status, not the badge color. An earlier version showed the red "Contact Support" card for any warning-looking state, which also caught harmless cases like a temporary polling timeout. We now only show the blocked treatment when the protocol actually blocked the payout. Timeouts keep the normal layout and just show a warning badge.
- Only show the date when we truly have a withdrawal timestamp. There is no on-chain "withdrawal submitted" time, and the vault's creation time is the peg-in time, which would be wrong. So the date is shown only once the vault provider has a record (which carries the real time), and is hidden for the brief moment before that.
- Stop the "in progress" spinner once nothing is actually progressing — including the case where polling gave up after repeated failures. This is handled by one shared helper so the rule is in a single place and is unit-tested.
- Keep the timing dynamic. The ~3 day estimate is computed from the contract's timelock parameter and live confirmations, not hardcoded. The "approximately 3 days" wording in the note is just friendly guidance shown next to the live estimate.

## Testing

- Unit tests for the stage mapping, the progress-bar math, and the "is it still in progress" helper (including the timeout edge case).
- Component tests for the new card: the right TX hash per stage and link gating, the challenge-period countdown and note, and the key rule that only a real block shows "Contact Support" and hides the progress bar while a timeout does not.
- Full vault test suite, lint, and type-check all pass.

## Still open (not blocking review)

- The "Read more about the withdrawal latency" and "Contact Support" links point at real, non-broken destinations (docs home and the Discord invite) but not the exact pages yet. There is a `TODO(product)` to swap in the final URLs.

Screenshots:

Screenshot_2026-06-05_13-53-28

Screenshot_2026-06-05_13-53-35

Screenshot_2026-06-05_13-53-41

### CODE REVIEW

File: services/vault/src/components/simple/PendingWithdrawSection.tsx [hunk 1]
```diff
@@ -121,99 +131,35 @@ function PendingWithdrawSectionContent({
...
-                      label={label}
-                      tooltip={tooltip}
-                    />
+                  vault={vault}
+                  pollingResult={pollingResult}
+                  timelockAssertBlocks={
+                    timelockAssert !== undefined
+                      ? Number(timelockAssert)
+                      : undefined
                   }
```
[the remark this answers is not in the document]
Comment (gbarkhatov) [reply services/vault/src/components/simple/PendingWithdrawSection.tsx]: done

**Comment by gbarkhatov** [comment]
Greptile "P1 - Do not hide withdrawals" also solved

File: services/vault/src/copy.ts [hunk 2]
```diff
@@ -638,18 +639,26 @@ export const COPY = {
...
-      // "Withdrawal" (not "Transaction hash") to distinguish from the deposit
-      // peg-in/Pre-Pegin row shown on the same pending-withdraw card.
-      label: "Withdrawal",
-      claimLabel: "Claim:",
-      assertLabel: "Assert:",
+    // Staged pending-withdraw card (Submitted → … → Payout sent / Blocked).
+    card: {
+      // Single withdrawal tx hash row: the claim tx, then the assert tx.
+      txHashLabel: "TX Hash",
+      // Shown in the TX Hash slot before the claim tx is broadcast.
```
Comment (jrwbabylonlab) [inline services/vault/src/copy.ts]: let's not use this term `tx hash`. it's not clear in code what this tx is, also more confusing for user.
We should be clear on the term. if this is the withdraw tx, show it as withdraw tx hash etc

File: services/vault/src/components/simple/PendingWithdrawCard.tsx [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+          claimerStatus={claimer?.status}
+        />
+      </VaultCardRow>
+
+      {/* Date — hidden until the VP has a withdrawal record (no earlier
+          withdrawal timestamp exists). */}
+      {timestampMs !== undefined && (
+        <VaultCardRow label="Date">
+          <span className="text-sm text-accent-primary">
+            {formatDateTime(new Date(timestampMs))}
```
Comment (jrwbabylonlab) [inline services/vault/src/components/simple/PendingWithdrawCard.tsx:207]: what's this date? is it the time when user initiated the withdraw? let's be clear on the term

**Comment by jrwbabylonlab** [comment]
I found it's confusing showing `2 days` while the text below it shows `will be unblocked in 3 days`.

What exactly is this countdown of 2 days mean? does it mean user just need to wait for 2 days for the funds to go back to their address? if it is, then why we show another text to explain the same thing?
If the countdown is only about `2 days for challenge period` then we should be clear on the term that this is not 2 days to withdraw, but rather one of the step which require 2 days to process. there are more steps to wait for.
You might need some input from UX designer on this. I think we should be clear on the information we are showing.

But of course, this PR is already making improvements on the existing UI.

File: services/vault/src/copy.ts [hunk 4]
```diff
@@ -638,18 +639,26 @@ export const COPY = {
...
+      // Live payout-eligibility estimate value (e.g. "~2 days").
+      estRemaining: (duration: string) => `~${duration}`,
+      // Challenge-period help note. The "~3 days" here is descriptive guidance;
+      // the live per-vault estimate is shown in the Est. Remaining row above it.
+      challengeNote:
+        "For your security, BTC will become unlocked and usable in approximately 3 days.",
+      learnMorePrefix: "Read more about the withdrawal latency ",
+      learnMoreLink: "here.",
+      // Error action on the Blocked stage.
+      contactSupport: "Contact Support",
```
Comment (jrwbabylonlab) [inline services/vault/src/copy.ts:679]: where we taking user to this `contact support`?

Comment (gbarkhatov) [reply services/vault/src/copy.ts:679]: discord

File: services/vault/src/copy.ts (hunk 2 above)
Comment (gbarkhatov) [reply services/vault/src/copy.ts]: done

**Comment by gbarkhatov** [comment]
@jrwbabylonlab 

C1 — Date → "Initiated" (copy.ts, PendingWithdrawCard.tsx)
The row label is now "Initiated" (the VP claimer-record timestamp ≈ when the withdrawal was initiated), and it's moved into COPY.pegout.card.initiatedLabel (no more inline literal).

C2 — Contact Support (constants.ts)
No code change needed — already points to the Babylon Discord invite. Tightened the comment to mark support as confirmed (kept the TODO only for the exact docs page).

C3 — "2 days" vs "3 days" (copy.ts, PendingWithdrawCard.tsx)

- The countdown row now reads "Challenge period ends → in ~2 days" (and "shortly" when elapsed, "Checking…" while loading) — so it's unambiguously the challenge-period step, not time-to-funds.
- The note no longer says "~3 days" (which conflicted with the live countdown); it now explains the step and that the payout follows: "…goes through an on-chain challenge period. After it ends, the payout is broadcast to your nominated address."
- Removed the now-unused payoutImminent copy.

File: services/vault/src/components/simple/PendingWithdrawCard.tsx (hunk 3 above)
Comment (gbarkhatov) [reply services/vault/src/components/simple/PendingWithdrawCard.tsx:207]: done

**Review by jrwbabylonlab — DISMISSED** [review]

**Review by jrwbabylonlab — APPROVED** [review]
